Bottom: 40a4c6e627833ef07fcb957a495e297a004f643a
Top:    8cbe8d4c9df94dab0c3ac0e520373087fc462395
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-25 16:05:35 +0100

Refresh of draw-c-separate-gathering-rout

---

diff --git a/src/draw.c b/src/draw.c
index 8b73644..68cb4a7 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -1172,11 +1172,19 @@ DrawElementPackageLowLevel (ElementTypePtr Element)
 void
 DrawVia (PinTypePtr Via)
 {
-  if (!Gathering)
-    SetPVColor (Via, VIA_TYPE);
-  DrawPinOrViaLowLevel (Via, true);
+  assert (Gathering);
+
+  GatherPV (Via);
+  GatherPVName (Via);
+}
+
+static void
+draw_pcb_via (PinTypePtr Via)
+{
+  SetPVColor (Via, VIA_TYPE);
+  draw_pcb_pv (Via, true);
   if (!TEST_FLAG (HOLEFLAG, Via) && TEST_FLAG (DISPLAYNAMEFLAG, Via))
-    DrawPinOrViaNameLowLevel (Via);
+    draw_pcb_pv_name (Via);
 }
 
 /* ---------------------------------------------------------------------------
